test/storage/index_builder_test.cpp

```diff
@@ -0,0 +1,142 @@
+#include "storage/index/index_builder.h"
+#include "storage/projected_row.h"
+#include "storage/sql_table.h"
+#include "test_util/catalog_test_util.h"
+#include "test_util/data_table_test_util.h"
+#include "test_util/random_test_util.h"
+#include "test_util/storage_test_util.h"
+#include "test_util/test_harness.h"
+#include "transaction/transaction_context.h"
+#include "transaction/transaction_manager.h"
+#include "type/type_id.h"
+#include "type/type_util.h"
+
+namespace terrier::storage::index {
+
+class IndexBuilderTests : public TerrierTest {
+ public:
+  catalog::Schema table_schema_;
+  catalog::IndexSchema index_schema_;
+
+  std::default_random_engine generator_;
+  const uint32_t num_threads_ = 4;
+
+  std::unique_ptr<DBMain> db_main_;
+  common::ManagedPointer<transaction::TransactionManager> txn_manager_;
+
+  // SqlTable
+  storage::SqlTable *sql_table_;
+  storage::ProjectedRowInitializer tuple_initializer_ =
+      storage::ProjectedRowInitializer::Create(std::vector<uint16_t>{1}, std::vector<uint16_t>{1});
+
+  common::WorkerPool thread_pool_{num_threads_, {}};
+
+ protected:
+  void SetUp() override {
+    thread_pool_.Startup();
+    db_main_ = terrier::DBMain::Builder().SetUseGC(true).SetUseGCThread(true).SetRecordBufferSegmentSize(1e6).Build();
+    txn_manager_ = db_main_->GetTransactionLayer()->GetTransactionManager();
+
+    auto idxcol = catalog::Schema::Column(
+        "attribute", type::TypeId::INTEGER, false,
+        parser::ConstantValueExpression(type::TransientValueFactory::GetNull(type::TypeId::INTEGER)));
+    auto nonidxcol = catalog::Schema::Column(
+        "attribute2", type::TypeId::INTEGER, false,
+        parser::ConstantValueExpression(type::TransientValueFactory::GetNull(type::TypeId::INTEGER)));
+    StorageTestUtil::ForceOid(&(idxcol), catalog::col_oid_t(1));
+    StorageTestUtil::ForceOid(&(nonidxcol), catalog::col_oid_t(2));
+    table_schema_ = catalog::Schema({idxcol, nonidxcol});
+    sql_table_ = new storage::SqlTable(db_main_->GetStorageLayer()->GetBlockStore(), table_schema_);
+    tuple_initializer_ = sql_table_->InitializerForProjectedRow({catalog::col_oid_t(1)});
+
+    std::vector<catalog::IndexSchema::Column> keycols;
+    keycols.emplace_back("", type::TypeId::INTEGER, false,
+                         parser::ColumnValueExpression(CatalogTestUtil::TEST_DB_OID, CatalogTestUtil::TEST_TABLE_OID,
+                                                       catalog::col_oid_t(1)));
+    StorageTestUtil::ForceOid(&(keycols[0]), catalog::indexkeycol_oid_t(1));
+    index_schema_ = catalog::IndexSchema(keycols, storage::index::IndexType::BWTREE, false, false, false, true);
+  }
+  void TearDown() override {
+    thread_pool_.Shutdown();
+
+    db_main_->GetTransactionLayer()->GetDeferredActionManager()->RegisterDeferredAction([=]() { delete sql_table_; });
+  }
+};
+
+/**
+ * This test creates multiple worker threads that all try to insert [0,num_inserts) as tuples in the table and into the
```

Comment:
wait really? seems like you're just scanning an empty table

Reply:
This was an old comment from some copied code. It will be deleted in a future update to this PR